internal/config/types.go

```diff
@@ -16,6 +16,7 @@ type Config struct {
 	ProcessMonitor ProcessMonitor `mapstructure:"process_monitor" yaml:"-"`
 	DataplaneAPI   DataplaneAPI   `mapstructure:"dataplane_api" yaml:"-"`
 	Client         Client         `mapstructure:"client" yaml:"-"`
+	Metrics        *Metrics       `mapstructure:"metrics" yaml:"-"`
```

Comment:
Should this be a pointer? Seems inconsistent with the other parameters

Reply:
It is inconsistent. The idea is that you could omit the `Metrics` config in the config file, which would mean that the metrics plugin would not start. We could use this same practice for the other plugins, but it sorta depends on how we want to go about this.

I suppose the alternative would be to have a list of plugins that we activate in the configuration. My instinct is to go with option #1, but you guys have more experience with this, what do you think? @dhurley, @oliveromahony

Reply:
We will have a separate discussion on this and align all the configurations to a similar style in a follow-up PR.